clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -67,4 +78,133 @@ abstract class IndexService[CI <: ClioIndex](
       elasticsearchIndex
     )
   }
+  private val keysToDrop =
+    Set(
+      ElasticsearchIndex.UpsertIdElasticsearchName,
+      ElasticsearchIndex.EntityIdElasticsearchName,
+    )
+
+  def validateUpsert(
+    indexKey: clioIndex.KeyType,
+    metadata: clioIndex.MetadataType,
+    force: Boolean
+  )(
+    implicit materializer: Materializer
+  ): Either[Exception, clioIndex.MetadataType] = {
+
+    // drop housekeeping and key fields (evaluate only metadata)
+    val dropKeys = keysToDrop ++ indexKey.getClass.getDeclaredFields
+      .map(field => ElasticsearchUtil.toElasticsearchName(field.getName))
+
+    val storedDocsForKey = Await
+      .result(
+        searchService
+          .simpleStringQueryMetadata(indexKey.asJson.pretty(Printer.noSpaces))(
```

Comment:
Ah, if we're JSON-ifying the key anyways then we don't need to use `FieldMapper` to get the names-to-strip-out. We can do something like:
```scala
val keyJson = indexKey.asJsonObject
val fieldsToDrop = keyJson.keys
```